openspp-project/openspp-registry
# Fix issue with issue in local registrar role unable to view assigned area in registry

## **Why is this change needed?**

- Fix the issue with this migrated module from Odoo 15 to 17, which causes the local registrars to access all registrants regardless of area.
- The spp.area model is also affected, which allows the access and assignment of areas not assigned to the local registrar.

## **How was the change implemented?**

- Override the following model functions to inject the domain filter that will limit the view of the areas assigned to the local registrars:
  - res.partner: web_search_read, search_read
  - spp.area: name_search, web_search_read, search_read

## **New unit tests**

## **Unit tests executed by the author**

## **How to test manually**

1. As the administrator, go to "Settings->Users and Groups".
2. Create a new user and assign the role "Local Registrar".
3. In the "Center Area" column select the area that will be assigned to the new local registrar.
4. Login as the new local registrar.
5. Open the "Registry UI (Groups or Individuals).
6. The list should have the registrants with "Center Area" equal to the area assigned to the local registrar.
7. When modifying the registrant's area, only the assigned area to the local registrar must be the only item in the drop-down list.

## **Related links**

[Issue in local registrar unable to view assigned area in registry](https://github.com/OpenSPP/openspp-modules/issues/531)
